File: src/Catalog.API/IntegrationEvents/Events/OrderStockItem.cs
namespace eShop.Catalog.API.IntegrationEvents.Events;

public record OrderStockItem
{
    public int ProductId { get; }
    public int Units { get; }

    public OrderStockItem(int productId, int units)
    {
        ProductId = productId;
        Units = units;
    }
}

File: src-java/eshop-integration-events/src/main/java/com/eshop/integrationevents/OrderStockItem.java
package com.eshop.integrationevents;

import com.fasterxml.jackson.annotation.JsonProperty;

public record OrderStockItem(
        @JsonProperty("ProductId") int productId,
        @JsonProperty("Units") int units) {

    public OrderStockItem {
        if (units < 0) {
            throw new IllegalArgumentException("units must not be negative");
        }
    }
}
